packages/teams-js/src/private/copilot.ts

```diff
@@ -41,12 +48,53 @@ export namespace copilot {
      *
      * @throws Error if {@linkcode app.initialize} has not successfully completed
      */
-    export function getEligibilityInfo(): AppEligibilityInformation {
+    export function getEligibilityInfo(): Promise<AppEligibilityInformation> {
       ensureInitialized(runtime);
       if (!isSupported()) {
         throw errorNotSupportedOnPlatform;
       }
-      return runtime.hostVersionsInfo!.appEligibilityInformation!;
+
+      // Return the eligibility information if it is already available
+      if (runtime.hostVersionsInfo?.appEligibilityInformation) {
+        return Promise.resolve(runtime.hostVersionsInfo!.appEligibilityInformation);
+      }
+
+      // Send message to host SDK to get eligibility information
+      return sendMessageToParentAsync<AppEligibilityInformation | SdkError>(
+        getApiVersionTag(copilotTelemetryVersionNumber, ApiName.Copilot_Eligibility_GetEligibilityInfo),
+        'copilot.eligibility.getEligibilityInfo',
```

Comment:
Can this be `ApiName.Copilot_Eligibility_GetEligibilityInfo` instead of a string

Reply:
I agree that hard coding is not awesome.

I don't think we should use `ApiName.Copilot_Eligibility_GetEligibilityInfo` here because that value is intended for telemetry use and could theoretically change without notice, causing the communication between teams-js and the host to break. 

Probably there should be an effort to use constants for the names of the functions that are passed to postMessage calls, but I'm not going to start that effort as part of this PR.

Reply:
I have been encouraging people in PRs to use the constant in both places since the string values are identical and every string you add makes the teamsjs package size bigger, so we should reuse where possible.

I understand your concern about the two values serving a different purpose but I think in practice that isn't likely to happen -- we'd have to change everything about the telemetry pipeline, reports, etc. if this string were to ever change. Minimizing package size increase outweighs the "different purpose" concern IMO.

Reply:
Discussed offline. I'm making the change here and opening a work item to track improving the code to prevent misuse of the telemetry constant vs a function name constant.